### ISSUE TITLE
[VO, Safari] Appendage positions are read out twice on each key press

### ISSUE THREAD

**Issue by terracoda** [phetsims/john-travoltage#237]
In version sim version 33,
MacOS Sierra 10.12.3 and VO, and 
Safari 10.0.3 , 
the positions of the appendages are read out twice. It is regular repetition. I personally do NOT find it terribly annoying.

This might be a VO bug?

**Comment by jessegreenberg** [comment]
Thanks @terracoda, I noticed this too, I am curious if this was happening in a previous version or if I broke something in a recent refactor.

**Comment by terracoda** [comment]
@jessegreenberg, I know it was happening at least on one prior version, but not sure if it re-surfaced here or not.

**Comment by jessegreenberg** [comment]
This is only happening in VO+Safari.  I have Safari 10.12.5 Beta on this device, and it is happening way back in version 1.3.0-dev.20. Before deploying that dev version for testing I tried the sim on our desired platforms and I don't remember seeing this issue.

**Comment by jessegreenberg** [comment]
Reverting to stable 10.12.4 for further testing.

**Comment by jessegreenberg** [comment]
This is happening in VO with Safari in this simple test case:

https://jsfiddle.net/gtbmqo2g/

**Comment by jessegreenberg** [comment]
Now VO is doing this in this example.

https://jsfiddle.net/gtbmqo2g/12/

All we are doing is updating aria-valuetext. Maybe VO is justing caching the JSFiddle example.

**Comment by jessegreenberg** [comment]
Restarting Safari and VO doesn't seem to help.

**Comment by jessegreenberg** [comment]
@jobara do you have any thoughts on this issue?  This seems like a bug with VO/Safari and ```aria-valuetext```.  Can you think of anything we are missing?

One alternative would be to stop using ```aria-valuetext``` and use ```aria-live``` to get the AT to read the value text exactly the way we want.

**Comment by jobara** [comment]
@jessegreenberg I couldn't see anything missing or think of any alternatives. It does seem like a VoiceOver bug. If you make the switch to `aria-live` you may want to check back periodically to remove the workaround when it is no longer needed. And just a reminder, you'll need to make sure that a new aria-live will behave as expected in regards to interruptions and etc.

**Comment by terracoda** [comment]
@jessegreenberg, I personally don't find the repetition too annoying - mostly I think because it is very regular, thus easy to ignore. More than one screen reader user has commented to me that they are accustomed to repetition. The repetition, however, could be more problematic when students are collaborating.

At this late date, we might just want to roll with the VO repetition, and send a bug report to Apple. @emily-phet, do you have an opinion here?

As @jobara mentioned, introducing aria-live at this point, may introduce other unexpected behaviour. Also, in Balloons, we have not yet been able to reliable hide the alerts in the PDOM.

**Comment by jessegreenberg** [comment]
Ok, thanks very much @jobara and @terracoda.  The problem is not just the repetition, when moving one position at a time the repeated ```aria-valuetext``` interrupts the ```aria-live``` charge alert that is already there.  On VO + Safari, the user could often never hear that charges are entering the body.

@emily-phet can you please give your opinion for the severity of this issue, and whether a workaround is necessary for the initial release?

**Comment by terracoda** [comment]
@jessegreenberg and @emily-phet, it is true that I do not hear the electron updates at regular intervals as the change which would be nice, but I do hear them. I didn't realize it was the position announcements that was interrupting them.

If the electron updates are essential information, they might need to have the role `alert` rather than `status`. But that might be too annoying.

**Comment by terracoda** [comment]
@jessegreenberg, would it be possible to dynamically change the politeness-level to assertive every 5 or 10 charges? Perhaps that would be a work around.

I understand from previous discussions with you that dynamically changing the role of an element is not allowed.

**Comment by jessegreenberg** [comment]
@terracoda we could do something like that, we wouldn't change the politeness level on an element but we could use a different element that is assertive every 5-10 charges.

**Comment by terracoda** [comment]
@jessegreenberg, and just a reminder that the charge information is available in the summary as well. It's ideal if the status alert for "Electrons on body" is read out during interaction, but if it gets bumped out of the cue, the information is still available outside of direct interaction.

**Comment by jessegreenberg** [comment]
Understood @terracoda, thanks.  Should I proceed with making the alert assertive every 5 charges?

**Comment by emily-phet** [comment]
@jessegreenberg If this workaround is on the of ~ an hour to implement, go for it. Otherwise, let's not address it for the upcoming publication.

**Comment by jessegreenberg** [comment]
Sounds good @emily-phet.  The workaround @terracoda proposed will take a couple of minutes and is much safer.  The workaround I proposed is much more error prone and will take a few hours to investigate whatever new issues come out of it, so I will proceed with what @terracoda suggested.

**Comment by emily-phet** [comment]
@jessegreenberg A safe fix that takes a few minutes to implement - wonderful!

**Comment by jessegreenberg** [comment]
@emily-phet @terracoda unfortunately it doesn't work as well as I hoped.  Sometimes it works great, sometimes the ```assertive``` update is still ignored when the value text is read twice, sometimes it triggers reading the alerts that were previously ignored, so the user might hear

- electrons on body: 5
- electrons on body: 2
- electrons on body: 3
- electrons on body: 4

And then all subsequent updates are treated as assertive.

**Comment by jessegreenberg** [comment]
Also, it looks like using ```aria-live=assertive``` is causing focus to behave weird just like ```role=status```.

@emily-phet @terracoda maybe we should bail on this issue and just accept this VoiceOver bug, what do you think?

**Comment by emily-phet** [comment]
@jessegreenberg Oh no! Yes, let's put a hold on this for now. We can come back to it later as we're getting ready to publish BASE in the summer.

**Comment by jessegreenberg** [comment]
Sounds good @emily-phet, marking as ```wontfix```.

**Comment by terracoda** [comment]
Totally agree.

**Comment by jessegreenberg** [comment]
We verified that this was a bug that has been fixed in Mojave and is no longer an issue for latest macOS and Safari. Closing.
